src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,73 @@
+package frc.robot.subsystems;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class IntakeSubsystem extends SubsystemBase {
+
+	// components
+	private CANSparkMax intakeMotor = new CANSparkMax(10, MotorType.kBrushless);
+	Solenoid leftSolenoid = new Solenoid(PneumaticsModuleType.CTREPCM, 2);
+  	Solenoid rightSolenoid = new Solenoid(PneumaticsModuleType.CTREPCM, 4);
+
+	// state variables
+	private boolean isExtended;
+	private int ballCount;
+  
+  public IntakeSubsystem() {
+		intakeMotor.setInverted(false);
+		// sets maximum rate at which the motor controller’s output is allowed to change
+		intakeMotor.setClosedLoopRampRate(0.3); 
+		// when motor stops, brake will not be activated. keep rolling until momentum runs out
+		intakeMotor.setIdleMode(IdleMode.kCoast); 
+		isExtended = false;
+  }
+
+	public void setIntake(double power) {
+		intakeMotor.set(power);
+	}
+
+	public void setBallCount(int count) {
+		ballCount = count;
+	}
+
+	public int getBallCount() {
+		return ballCount;
+	}
+
+	public void extendIntake() {
+		leftSolenoid.set(true);
+		rightSolenoid.set(true);
+		isExtended = true; // update state
+	}
+
+	public void contractIntake() {
+		leftSolenoid.set(false);
+		rightSolenoid.set(false);
+		isExtended = false; // update state
+	}
+
+	public void setIsExtended(Boolean state) {
+		isExtended = state;
+	}
+
+	public Boolean getIsExtended() {
+		return isExtended;
+	}
+
+	// TODO is DisabledCommand depracated or something? Why does the import not work
```

Comment:
DisabledCommand is not part of the framework. I came up with it last year and made it run in Robot.java. However, we realized we should not do that because it seems to interfere with the robot's communication with the driver station.